static/app/utils/profiling/hooks/useAggregateFlamegraphQuery.ts

```diff
@@ -1,15 +1,23 @@
 import {useMemo} from 'react';
 
 import {normalizeDateTimeParams} from 'sentry/components/organizations/pageFilters/parse';
+import {PageFilters} from 'sentry/types';
 import {useApiQuery} from 'sentry/utils/queryClient';
 import {MutableSearch} from 'sentry/utils/tokenizeSearch';
 import useOrganization from 'sentry/utils/useOrganization';
-import usePageFilters from 'sentry/utils/usePageFilters';
 
-export function useAggregateFlamegraphQuery({transaction}: {transaction: string}) {
+export function useAggregateFlamegraphQuery({
+  projects,
+  datetime,
+  environments,
+  transaction,
+}: {
+  datetime: Partial<PageFilters['datetime']>;
+  environments: string[];
+  projects: number[];
+  transaction: string;
+}) {
   const organization = useOrganization();
-  const {selection} = usePageFilters();
```

Comment:
you could leave the selection as a optional prop to the hook and use the values from the page filters as the default to simplify some things

Reply:
I thought about that, but then it just makes it semi-implicit and w/o a way to disable certain params (unless we reserve null or something as a value). I'd rather keep this explicit since it's not that many lines of code